Code/GraphMol/FileParsers/MultithreadedMolSupplier.cpp

```diff
@@ -0,0 +1,91 @@
+//
+//  Copyright (C) 2020 Shrey Aryan
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include "MultithreadedMolSupplier.h"
+
+#include <GraphMol/SmilesParse/SmilesWrite.h>
+
+namespace RDKit {
+
+//! method for thread-safe printing, only for debugging
+struct PrintThread : public std::stringstream {
+  static inline std::mutex cout_mutex;
+  ~PrintThread() {
+    std::lock_guard<std::mutex> l{cout_mutex};
+    std::cout << rdbuf();
+  }
+};
+
+void MultithreadedMolSupplier::reader() {
+  std::string record;
+  unsigned int lineNum;
+  while (extractNextRecord(record, lineNum)) {
+    auto r = std::tuple<std::string, int>{record, lineNum};
```

Comment:
I think we should also add the record number to this tuple.
It's relatively easy to figure that out from the line number when using some SMILES files, but much more difficult with SDF.